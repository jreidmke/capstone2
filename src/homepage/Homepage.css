#night {
    width: 100%;
}

#bill-directions {
    padding-top: 10em;
}

#bill-row {
    padding-top: 50em;
}

#link-img {
    height: 20em;
    width: 30em;
    opacity: .5;
    border-radius: 10%;
}

#link-img:hover {
    opacity: 1;
}

#image-row {
    padding-top: 3em;
}

#pol-search-row {
    padding-top: 4em;
}