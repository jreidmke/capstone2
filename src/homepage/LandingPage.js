import Image from 'react-bootstrap/Image'; 
import './Landingpage.css';
import capImg from './../images/capitol-image.jpg'; 
import conImg from './../images/old-congress.jpg';
import modImg from './../images/modern-congress.jpg'; 
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row'
import Col from 'react-bootstrap/Col'; 
import { FaArrowDown } from 'react-icons/fa'; 
import { GiCongress } from 'react-icons/gi'; 
import {Link} from 'react-router-dom'; 

const LandingPage = () => {
    return(
        <Container id='landingpage-container'>

            <Row id='top-row'>
                <Col>
                    <h1 className='mt-2'>Welcome to Article 1</h1>
                    <GiCongress size='5em'/>
                    <hr/>
                    <h3>Your one stop shop for info on the Bills, Senators and Representatives that make up the U.S. Capitol.</h3>
                </Col>
                <Col>
                    <Image src={capImg} id='landingpage-img'/>
                </Col>
            </Row>

            <Row>
                <Col>
                    <FaArrowDown size='10em' color='white'/>
                </Col>
            </Row>

            <Row id='how-to-row'>
                <Col>
                    <Image src={conImg} id='landingpage-img'/>
                </Col>
                <Col>
                    <h1>Search by Bill or by Politician</h1>
                    <hr/>
                    <h3>Article 1 uses multiple, up-to-date API's to supply you with what's happening right now in Congress. Read about how your Senator voted on the latest controversial bill. Or see who donated to their campaign!</h3>
                </Col>
            </Row>

            <Row>
                <Col>
                    <FaArrowDown size='10em' color='white'/>
                </Col>
            </Row>

            <Row id='bottom-row'>
                <Col>
                    <h1>Keep Up to Date with Reps and Senators Across the U.S.</h1>
                    <hr/>
                    <h3>Use our location-based search to discover who represents the constituents of the towns, cities and rural spaces that make up America.</h3>
                </Col>
                <Col>
                    <Image src={modImg} id='landingpage-img'/>
                </Col>
            </Row>

            <Row>
                <Col>
                    <a href='/home'><GiCongress size='10em' id='bottom-icon'/></a>
                </Col>
            </Row>

            <Row className='my-5'>
                <Col>
                    <h4>Click the Capitol Building to Start</h4>
                </Col>
            </Row>

        </Container>
    )
}

export default LandingPage;
