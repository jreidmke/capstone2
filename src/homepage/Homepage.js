import RecentBillCard from './../recent-bills/RecentBillCard';
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row'
import Col from 'react-bootstrap/Col'; 
import './Homepage.css'; 
import BillForm from './../searches/BillForm';
import SimpleMap from '../map/Map';
import chamberImage from './../images/chamberImage.jpg';
import constitutionImage from './../images/constitutionImage.jpg';
import { FaArrowLeft, FaArrowDown } from 'react-icons/fa'; 
import * as Scroll from 'react-scroll';
import CustomAlert from './../common/Alert';


const Hompage = () => {
    return(

        <Container>
        <CustomAlert color='danger' message='Nomination Data is Currently Down! We apologize for the inconvenience.'/>

            <Row className='my-5'>
                <Col><h2>To begin, select what you want to search!</h2></Col>
            </Row>

            <Row>
                
                <Col><FaArrowDown size='5em' color='white'/></Col>
                <Col><FaArrowDown size='5em' color='white'/></Col>
            </Row>

            <Row id='image-row'>
                <Col>

                    <Scroll.Link activeClass="active" className="pol-search-link" to="politician-search" spy={true} smooth={true} duration={500}>
                            <h4>I want to search <b>Politicians</b>.</h4>
                            <img src={chamberImage} id='link-img'/>
                    </Scroll.Link>
                </Col>

                <Col>
                    <Scroll.Link activeClass="active" className="bill-search-link" to="bill-search" spy={true} smooth={true} duration={500}>
                        <h4>I want to search <b>Bills</b>.</h4>
                        <img src={constitutionImage} id='link-img'/>
                    </Scroll.Link>
                </Col>
            </Row>

            <Row>

            </Row>


            <br/><br/>
            <br/><br/>
            <br/><br/>
            <br/><br/>
            <br/><br/>
            <br/><br/>
            <br/><br/>
            <br/><br/>
            <br/><br/>
            <br/><br/>
            <br/><br/>
            
                <Scroll.Element name="politician-search" className="element">
                    <Row id='pol-search-row'>
                        <SimpleMap/>
                    </Row>
                </Scroll.Element>   

                <Row id='bill-row'>    
                    <Scroll.Element name="bill-search" className="element">
                        <Row>
                            <Col>
                                <RecentBillCard/>
                            </Col>
                            <Col>
                                <Row id='bill-directions'>
                                    <h3>To search bills, simply enter a term (ex: gun control or entitlement reform) and submit. We'll look thru our database for any related bills.</h3>
                                    <BillForm isNavbar={false}/>
                                    
                                    <h3>...need some inspiration? Take a look at some recently discussed bills right here!</h3>
                                    <Col><FaArrowLeft className='text-center' size='10em'/></Col>
                                </Row>
                            </Col>
                            
                        </Row>
                    </Scroll.Element>
                </Row>

        </Container>
    )
}



export default Hompage; 

            // {/* <Tabs defaultActiveKey='politicians' mountOnEnter>
            //     <Tab eventKey='politicians' title='Search Politicians'> */}
            //     <Scroll.Element name="test1" className="element">
            //         <Row>
            //             <SimpleMap/>
            //         </Row>
            //     </Scroll.Element>
            //     {/* </Tab>
            //     <Tab eventKey='bills' title='Search Bills'> */}
            //         <Row id='bill-row'>    
            //             <Col>
            //                 <Row id='bill-directions'>
            //                     <h3>To search bills, simply enter a term and submit. We'll look thru our database for any related bills.</h3>
            //                     <BillForm/>
            //                     <h3>Need some inspiration? Take a look at some recently discussed bills right here!</h3>
            //                     <Col><FaArrowRight className='text-center' size='10em'/></Col>
            //                 </Row>
            //             </Col>
            //             <Col>
            //                 <RecentBillCard/>
            //             </Col>
            //         </Row>
            //     {/* </Tab>
            // </Tabs> */}



            {/* <Row>
                
                <Col>

                    <Row id='address-search-row'>
                        <Col>
                            <h2>Search For a Politician<Modal message={addressSearch}/></h2>
                            <AddressSearch/>
                        </Col>
                    </Row>

                    <Row id='bill-search-row'>
                        <Col>
                            <h2>Search For a Bill<Modal message={billSearch}/></h2>
                            <BillForm/>
                        </Col>
                    </Row>
                    
                </Col>
                <Col>
                    <h2>Recent Bills</h2>
                    <RecentBillCard/>
                </Col>
            </Row> */}


            
            // <Row>
            //     <Col>
            //         <Row>
            //             <h2>I want to learn about politicians.</h2>
            //         </Row>

            //         <Row>
            //             <Col>
            //                 <FaArrowDown size='10em' color='white'/>
            //             </Col>
            //         </Row>
            //     </Col>
{/* <Row>
                <Col>
                    <FaArrowDown size='10em' color='white'/>
                </Col>
            </Row> */}

            //     <Col><h2>I want to read about bills.</h2></Col>
            // </Row>