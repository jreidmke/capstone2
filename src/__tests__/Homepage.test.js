import React from "react";
import { render } from "@testing-library/react";
import Homepage from './../homepage/Homepage';
import LandingPage from './../homepage/LandingPage';

test('renders without crashing', function() {
    render(<Homepage/>); 
})

test('it matches snapshot', function() {
    const {asFragment} = render(
        <Homepage/>
    );
    expect(asFragment()).toMatchSnapshot();
}); 

test('renders without crashing', function() {
    render(<LandingPage/>); 
})

test('it matches snapshot', function() {
    const {asFragment} = render(
        <LandingPage/>
    );
    expect(asFragment()).toMatchSnapshot();
}); 