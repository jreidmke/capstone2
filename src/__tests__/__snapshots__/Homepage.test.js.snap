// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="fade alert alert-danger alert-dismissible show"
      role="alert"
    >
      <button
        class="close"
        type="button"
      >
        <span
          aria-hidden="true"
        >
          Ã—
        </span>
        <span
          class="sr-only"
        >
          Close alert
        </span>
      </button>
      <b>
        Nomination Data is Currently Down! We apologize for the inconvenience.
      </b>
    </div>
    <div
      class="my-5 row"
    >
      <div
        class="col"
      >
        <h2>
          To begin, select what you want to search!
        </h2>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col"
      >
        <svg
          color="white"
          fill="currentColor"
          height="5em"
          stroke="currentColor"
          stroke-width="0"
          style="color: white;"
          viewBox="0 0 448 512"
          width="5em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M413.1 222.5l22.2 22.2c9.4 9.4 9.4 24.6 0 33.9L241 473c-9.4 9.4-24.6 9.4-33.9 0L12.7 278.6c-9.4-9.4-9.4-24.6 0-33.9l22.2-22.2c9.5-9.5 25-9.3 34.3.4L184 343.4V56c0-13.3 10.7-24 24-24h32c13.3 0 24 10.7 24 24v287.4l114.8-120.5c9.3-9.8 24.8-10 34.3-.4z"
          />
        </svg>
      </div>
      <div
        class="col"
      >
        <svg
          color="white"
          fill="currentColor"
          height="5em"
          stroke="currentColor"
          stroke-width="0"
          style="color: white;"
          viewBox="0 0 448 512"
          width="5em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M413.1 222.5l22.2 22.2c9.4 9.4 9.4 24.6 0 33.9L241 473c-9.4 9.4-24.6 9.4-33.9 0L12.7 278.6c-9.4-9.4-9.4-24.6 0-33.9l22.2-22.2c9.5-9.5 25-9.3 34.3.4L184 343.4V56c0-13.3 10.7-24 24-24h32c13.3 0 24 10.7 24 24v287.4l114.8-120.5c9.3-9.8 24.8-10 34.3-.4z"
          />
        </svg>
      </div>
    </div>
    <div
      class="row"
      id="image-row"
    >
      <div
        class="col"
      >
        <a
          class="pol-search-link"
        >
          <h4>
            I want to search 
            <b>
              Politicians
            </b>
            .
          </h4>
          <img
            id="link-img"
            src="chamberImage.jpg"
          />
        </a>
      </div>
      <div
        class="col"
      >
        <a
          class="bill-search-link"
        >
          <h4>
            I want to search 
            <b>
              Bills
            </b>
            .
          </h4>
          <img
            id="link-img"
            src="constitutionImage.jpg"
          />
        </a>
      </div>
    </div>
    <div
      class="row"
    />
    <br />
    <br />
    <br />
    <br />
    <br />
    <br />
    <br />
    <br />
    <br />
    <br />
    <br />
    <br />
    <br />
    <br />
    <br />
    <br />
    <br />
    <br />
    <br />
    <br />
    <br />
    <br />
    <div
      class="element"
      name="politician-search"
    >
      <div
        class="row"
        id="pol-search-row"
      >
        <div>
          Loading...
        </div>
      </div>
    </div>
    <div
      class="row"
      id="bill-row"
    >
      <div
        class="element"
        name="bill-search"
      >
        <div
          class="row"
        >
          <div
            class="col"
          >
            <div
              class="card"
            >
              <nav
                class="nav card-header-tabs nav-tabs"
                role="tablist"
              >
                <a
                  aria-selected="false"
                  class="nav-item nav-link"
                  data-rb-event-key="introduced"
                  href="#"
                  role="tab"
                >
                  Introduced
                </a>
                <a
                  aria-selected="true"
                  class="nav-item nav-link active"
                  data-rb-event-key="active"
                  href="#"
                  role="tab"
                >
                  Active
                </a>
                <a
                  aria-selected="false"
                  class="nav-item nav-link"
                  data-rb-event-key="passed"
                  href="#"
                  role="tab"
                >
                  Passed
                </a>
                <a
                  aria-selected="false"
                  class="nav-item nav-link"
                  data-rb-event-key="enacted"
                  href="#"
                  role="tab"
                >
                  Enacted
                </a>
              </nav>
              <div
                class="tab-content"
              >
                <div
                  aria-hidden="false"
                  class="fade tab-pane active show"
                  role="tabpanel"
                >
                  <div
                    class="card-body"
                  >
                    <div
                      class="list-group"
                    >
                      <div
                        class="spinner-border"
                        role="status"
                      >
                        <span
                          class="sr-only"
                        >
                          Loading...
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="col"
          >
            <div
              class="row"
              id="bill-directions"
            >
              <h3>
                To search bills, simply enter a term (ex: gun control or entitlement reform) and submit. We'll look thru our database for any related bills.
              </h3>
              <form>
                <input
                  class="isNotNavbar"
                  id="term"
                  name="term"
                  placeholder="Health Care"
                  type="text"
                  value=""
                />
                <button
                  class="mb-3 ml-2 btn btn-info"
                  type="button"
                >
                  Submit
                </button>
                <br />
              </form>
              <h3>
                ...need some inspiration? Take a look at some recently discussed bills right here!
              </h3>
              <div
                class="col"
              >
                <svg
                  class="text-center"
                  fill="currentColor"
                  height="10em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 448 512"
                  width="10em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`it matches snapshot 2`] = `
<DocumentFragment>
  <div
    class="container"
    id="landingpage-container"
  >
    <div
      class="row"
      id="top-row"
    >
      <div
        class="col"
      >
        <h1
          class="mt-2"
        >
          Welcome to Article 1
        </h1>
        <svg
          fill="currentColor"
          height="5em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="5em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M247.002 32v47.725a56.476 56.476 0 0 1 17.996 0V32h-17.996zM256 96.998c-11.5 0-23.002 5.001-23.002 15.002v10.1c7.37-2 15.05-3.098 23.002-3.098 7.953 0 15.632 1.098 23.002 3.098V112c0-10-11.502-15.002-23.002-15.002zm0 40c-28.25 0-53.982 17.938-72.867 42.488-16.345 21.249-26.934 47.277-29.496 67.516h204.726c-2.562-20.239-13.15-46.267-29.496-67.516-18.885-24.55-44.617-42.488-72.867-42.488zm-80 78.004h32v17.996h-32v-17.996zm64 0h32v17.996h-32v-17.996zm64 0h32v17.996h-32v-17.996zm-170.29 49.996l-7.53 16.004h259.64l-7.53-16.004H133.71zm1.292 34v28.004h17.996v-28.004h-17.996zm56 0v28.004h17.996v-28.004h-17.996zm56 0v28.004h17.996v-28.004h-17.996zm56 0v28.004h17.996v-28.004h-17.996zm56 0v28.004h17.996v-28.004h-17.996zm-270.004 46v16.004h334.004v-16.004H88.998zm14.004 34v60.004h17.996v-60.004h-17.996zm48 0v60.004h17.996v-60.004h-17.996zm48 0v60.004h17.996v-60.004h-17.996zm48 0v60.004h17.996v-60.004h-17.996zm48 0v60.004h17.996v-60.004h-17.996zm48 0v60.004h17.996v-60.004h-17.996zm48 0v60.004h17.996v-60.004h-17.996zm-350.004 78v30.004h430.004v-30.004H40.998z"
          />
        </svg>
        <hr />
        <h3>
          Your one stop shop for info on the Bills, Senators and Representatives that make up the U.S. Capitol.
        </h3>
      </div>
      <div
        class="col"
      >
        <img
          class=""
          id="landingpage-img"
          src="capitol-image.jpg"
        />
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col"
      >
        <svg
          color="white"
          fill="currentColor"
          height="10em"
          stroke="currentColor"
          stroke-width="0"
          style="color: white;"
          viewBox="0 0 448 512"
          width="10em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M413.1 222.5l22.2 22.2c9.4 9.4 9.4 24.6 0 33.9L241 473c-9.4 9.4-24.6 9.4-33.9 0L12.7 278.6c-9.4-9.4-9.4-24.6 0-33.9l22.2-22.2c9.5-9.5 25-9.3 34.3.4L184 343.4V56c0-13.3 10.7-24 24-24h32c13.3 0 24 10.7 24 24v287.4l114.8-120.5c9.3-9.8 24.8-10 34.3-.4z"
          />
        </svg>
      </div>
    </div>
    <div
      class="row"
      id="how-to-row"
    >
      <div
        class="col"
      >
        <img
          class=""
          id="landingpage-img"
          src="old-congress.jpg"
        />
      </div>
      <div
        class="col"
      >
        <h1>
          Search by Bill or by Politician
        </h1>
        <hr />
        <h3>
          Article 1 uses multiple, up-to-date API's to supply you with what's happening right now in Congress. Read about how your Senator voted on the latest controversial bill. Or see who donated to their campaign!
        </h3>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col"
      >
        <svg
          color="white"
          fill="currentColor"
          height="10em"
          stroke="currentColor"
          stroke-width="0"
          style="color: white;"
          viewBox="0 0 448 512"
          width="10em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M413.1 222.5l22.2 22.2c9.4 9.4 9.4 24.6 0 33.9L241 473c-9.4 9.4-24.6 9.4-33.9 0L12.7 278.6c-9.4-9.4-9.4-24.6 0-33.9l22.2-22.2c9.5-9.5 25-9.3 34.3.4L184 343.4V56c0-13.3 10.7-24 24-24h32c13.3 0 24 10.7 24 24v287.4l114.8-120.5c9.3-9.8 24.8-10 34.3-.4z"
          />
        </svg>
      </div>
    </div>
    <div
      class="row"
      id="bottom-row"
    >
      <div
        class="col"
      >
        <h1>
          Keep Up to Date with Reps and Senators Across the U.S.
        </h1>
        <hr />
        <h3>
          Use our location-based search to discover who represents the constituents of the towns, cities and rural spaces that make up America.
        </h3>
      </div>
      <div
        class="col"
      >
        <img
          class=""
          id="landingpage-img"
          src="modern-congress.jpg"
        />
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col"
      >
        <a
          href="/home"
        >
          <svg
            fill="currentColor"
            height="10em"
            id="bottom-icon"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="10em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M247.002 32v47.725a56.476 56.476 0 0 1 17.996 0V32h-17.996zM256 96.998c-11.5 0-23.002 5.001-23.002 15.002v10.1c7.37-2 15.05-3.098 23.002-3.098 7.953 0 15.632 1.098 23.002 3.098V112c0-10-11.502-15.002-23.002-15.002zm0 40c-28.25 0-53.982 17.938-72.867 42.488-16.345 21.249-26.934 47.277-29.496 67.516h204.726c-2.562-20.239-13.15-46.267-29.496-67.516-18.885-24.55-44.617-42.488-72.867-42.488zm-80 78.004h32v17.996h-32v-17.996zm64 0h32v17.996h-32v-17.996zm64 0h32v17.996h-32v-17.996zm-170.29 49.996l-7.53 16.004h259.64l-7.53-16.004H133.71zm1.292 34v28.004h17.996v-28.004h-17.996zm56 0v28.004h17.996v-28.004h-17.996zm56 0v28.004h17.996v-28.004h-17.996zm56 0v28.004h17.996v-28.004h-17.996zm56 0v28.004h17.996v-28.004h-17.996zm-270.004 46v16.004h334.004v-16.004H88.998zm14.004 34v60.004h17.996v-60.004h-17.996zm48 0v60.004h17.996v-60.004h-17.996zm48 0v60.004h17.996v-60.004h-17.996zm48 0v60.004h17.996v-60.004h-17.996zm48 0v60.004h17.996v-60.004h-17.996zm48 0v60.004h17.996v-60.004h-17.996zm48 0v60.004h17.996v-60.004h-17.996zm-350.004 78v30.004h430.004v-30.004H40.998z"
            />
          </svg>
        </a>
      </div>
    </div>
    <div
      class="my-5 row"
    >
      <div
        class="col"
      >
        <h4>
          Click the Capitol Building to Start
        </h4>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
